### PR TITLE
Keep global registry of all visualizers & context systems instead of isolating per space view class

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Refactor towards #4490

This will allow us to store additional state with each registry entry: This will be used to register a `VisualizerEntitySubscriber` and store a handle to it for every visualizer.

All the "interesting" changes are in `crates/re_viewer_context/src/space_view/space_view_class_registry.rs`, everything else is just ripples.

Note that this pr sneakily starts the renaming of `ViewPart(System)` to just `Visualizer`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Full build: [app.rerun.io](https://app.rerun.io/pr/4540/index.html)
  * Partial build: [app.rerun.io](https://app.rerun.io/pr/4540/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json) - Useful for quick testing when changes do not affect examples in any way
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4540)
- [Docs preview](https://rerun.io/preview/13e2dbf32f0dd151e7ce18c2b98af9f1f929eb63/docs) 
- [Examples preview](https://rerun.io/preview/13e2dbf32f0dd151e7ce18c2b98af9f1f929eb63/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Comment by teh-cmc** [comment]
Putting a massive renaming change in the same commit as everything else was not very nice of you @Wumpf :smiling_face_with_tear:

**Review by teh-cmc — APPROVED** [review]
Im too far removed from all of this to really discuss the semantics of it all, but understood the general idea and didn't see anything suprising so... :green_circle:

**Comment by Wumpf** [comment]
> Putting a massive renaming change

ah yeah it's not the massive change that is going to happen. But still kinda guilty as I started chipping away on it.

File: crates/re_viewer_context/src/space_view/space_view_class_registry.rs [hunk 1]
```diff
@@ -64,90 +71,76 @@ impl SpaceViewSystemRegistry {
...
-        }
-    }
 }
 
-/// Entry in [`SpaceViewClassRegistry`]
+/// Space view class entry in [`SpaceViewClassRegistry`].
 struct SpaceViewClassRegistryEntry {
     class: Box<dyn DynSpaceViewClass>,
-    systems: SpaceViewSystemRegistry,
+    context_systems: HashSet<ViewSystemIdentifier>,
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/re_viewer_context/src/space_view/space_view_class_registry.rs:115]: hehe :D. Was actually wondering if it's still `VisualizerSystem` in this context, to not make system stick out too much

**Pull request merged**